cmd/podman/port.go

```diff
@@ -0,0 +1,145 @@
+		Description: portDescription,
+		Flags:       portFlags,
+		Action:      portCmd,
+		ArgsUsage:   "CONTAINER-NAME [mapping]",
+	}
+)
+
+func portCmd(c *cli.Context) error {
+	var userProto, containerName string
+	var userPort int
+	var container *libpod.Container
+	var containers []*libpod.Container
+
+	args := c.Args()
+	if err := validateFlags(c, portFlags); err != nil {
+		return err
+	}
+
+	if c.Bool("latest") && c.Bool("all") {
+		return errors.Errorf("the 'all' and 'latest' options cannot be used together")
+	}
+	if c.Bool("all") && len(args) > 0 {
+		return errors.Errorf("no additional arguments can be used with 'all'")
+	}
+	if len(args) == 0 && !c.Bool("latest") && !c.Bool("all") {
+		return errors.Errorf("you must supply a running container name or id")
+	}
+	if !c.Bool("latest") && !c.Bool("all") {
+		containerName = args[0]
+	}
+
+	port := ""
+	if len(args) > 1 && !c.Bool("latest") {
+		port = args[1]
+	}
+	if len(args) == 1 && c.Bool("latest"){
+		port = args[0]
+	}
+	if port != "" {
+		fields := strings.Split(port, "/")
+		// User supplied at least port
+		var err error
+		if len(fields) >= 1 {
+			p := fields[0]
+			userPort, err = strconv.Atoi(p)
+			if err != nil {
+				return errors.Wrapf(err, "unable to format port")
+			}
+
+		}
+		// User supplied port and protocol
+		if len(fields) == 2 {
+			userProto = fields[1]
+		}
+		// Format is incorrect
+		if len(fields) > 2 || len(fields) < 1 {
+			return errors.Errorf("port formats are port/protocol. '%s' is invalid", port)
+		}
+	}
+
+	runtime, err := getRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	if !c.Bool("latest") && !c.Bool("all") {
+		container, err = runtime.LookupContainer(containerName)
+		if err != nil {
+			return errors.Wrapf(err, "unable to find container %s", containerName)
+		}
+		containers = append(containers, container)
+	} else if c.Bool("latest") {
+		container, err = runtime.GetLatestContainer()
+		containers = append(containers, container)
+	} else {
+		containers, err = runtime.GetRunningContainers()
+		if err != nil {
+			return errors.Wrapf(err, "unable to get all containers")
+		}
+	}
+
+	for _, con := range containers {
+		if state, _ := con.State(); state != libpod.ContainerStateRunning {
+			// If the container is not running, return silently
+			return nil
+		}
+		if c.Bool("all")	{
+			fmt.Println(con.ID())
+		}
+		// Iterate mappings
+		for _, v := range con.Config().PortMappings {
+			hostIP := v.HostIP
+			// Set host IP to 0.0.0.0 if blank
+			if hostIP == "" {
+				hostIP = "0.0.0.0"
+			}
+			// If not searching by port or port/proto, then dump what we see
+			if port == "" {
+				fmt.Printf("%d/%s -> %s:%d\n", v.ContainerPort, v.Protocol, hostIP, v.HostPort)
+				continue
+			}
+			// We have a match on ports
+			if v.ContainerPort == int32(userPort) {
+				if userProto == "" || userProto == v.Protocol {
+					fmt.Printf("%s:%d", hostIP, v.HostPort)
+					break
```

Comment:
Why is this a break?  Shouldn't it continue going through the port mappings?

Reply:
nope, should only be one so it can stop its iteration.